the-collab-lab/tcl-74-smart-shopping-list
# Issue #15: Implement IconButton and update icon imports for consistent design and functionality

## Description

This PR addresses enhancements to improve the overall design and functionality of the application. 

- All icons are imported from the "react-icons" library, replacing CDN-imported icons. 
- All buttons and icons incorporate IconButton components. 
- The share list icon has been moved to be a sibling of the disclosure header, rather than being integrated within it.
- A list name is displayed on the `manage-list` page. 
- The purchase frequency options for items are now presented as vertically aligned radio buttons.

## Related Issue

- Closes #40 

## Acceptance Criteria

- [x]  Replace all CDN icon imports with imports from "react-icons."
- [x]  Ensure the share button is positioned as a sibling of the disclosure.
- [x]  Implement IconButton for all applicable buttons and links, including:

  - [x]  Add (Create List) button
  - [x]  Share button
  - [x]  Sign-in button
  - [x]  View lists button in footer
  - [x]  Add item button in footer
  - [x]  Sign out button in footer
  - [x]  Delete button within the disclosure, with refactoring to support keyboard accessibility.

## Type of Changes
`enhancement`
 `bug fix`
`refactoring` 

## Updates

### Before



### After



## Testing Steps / QA Criteria

- Confirm that your List Name is now displayed in the header of `manage-list` page. 
- Confirm that all buttons function as expected.

  For `Landing page`
  - Sign in button
  
  For `View List` page: 
  - Add a button for creating a list. 
  - Share button for each list.
  - Delete button for each item.
  
  For `Manage-List` page
  - Add a button for creating an item on a list.
  - Share button for sharing with a recipient. 
  
  For Footer
  - View List button
  - Add Items button
  - Sign out button
